### PR TITLE
Change FollowersController#remove to use section_code.

### PR DESCRIPTION
**Pull request by ashercodeorg**
### IMAGES

ONE SECTION, ACCOUNT w/ PASSWORD:
image

MULTIPLE SECTIONS, ACCOUNT w/ PASSWORD:
image

ONE SECTION, ACCOUNT w/o PASSWORD:
image

### RATIONALE

There are several motivations for this PR. As we no longer index uniquely on teacher and student, it does not make sense to delete on teacher and student. As we want to unblock the co-teachers feature, we need to delete on section rather than (one) teacher.

### INDEXING

Note that this PR requires an index on `section_id` and `student_user_id`. This was done in #13999 and DTPed in #14028.

### CODE REVIEW

**Comment by ashercodeorg** [comment]
@ryansloan: Do the images above look (read) as expected? Do we want to be including the section name and/or the section code?

**Comment by ashercodeorg** [comment]
Note that the CircleCI failure is legitimate. Is there an error in the PR causing the translation to show up as missing? If so, what? If not, how do I go about fixing the failure?

```
Failure:
test_remove_has_nice_error_when_student_does_not_actually_have_teacher(FollowersControllerTest) [/home/ubuntu/code-dot-org/dashboard/test/controllers/followers_controller_test.rb:310]:
--- expected
+++ actual
@@ -1 +1 @@
-"The section could not be found."
+"translation missing: en-US.follower.error.section_not_found(section_code: params[:section_code])"
```
(ignore the diff in the actual text, that I'm aware of)

**Comment by ryansloan** [comment]
LGTM

File: dashboard/app/controllers/followers_controller.rb [hunk 1]
```diff
@@ -14,21 +14,36 @@ def create
...
+    if @section
+      f = Follower.where(section: @section.id, student_user_id: current_user.id).first
+    end
 
-    unless f.present?
-      redirect_to root_path, alert: t('teacher.user_not_found')
+    unless @section && f
+      # TODO(asher): Change the alert message to section.
+      redirect_to root_path, alert: t(
+        'follower.error.section_not_found(section_code: params[:section_code])'
```
Comment (Hamms) [inline dashboard/app/controllers/followers_controller.rb]: shouldn't this be 

```
t(
  'follower.error.section_not_found',
  section_code: params[:section_code]
)
```

? I believe this is the source of the test failure

File: dashboard/app/views/home/index.html.haml [hunk 2]
```diff
@@ -57,21 +57,20 @@
...
-            -if current_user.encrypted_password?
-              = raw current_user.teachers.map{|t| t[:name] + ' ' + link_to(t('teacher.remove'), remove_follower_path(teacher_user_id: t[:id], student_user_id: current_user.id), confirm: t('teacher.confirm.remove'), method: :post)}.join(', ')
-
-            -else
-              = raw current_user.teachers.map{|t| t[:name]}.join(",")
+            = t('section.your_section', count: current_user_sections_as_student.length)
+            - if current_user.encrypted_password?
+              = raw current_user_sections_as_student.map {|section| t('section.code_with_teacher', section_code: section.code, teacher_name: section.teacher.name) + ' ' + link_to(t('section.remove'), remove_follower_path(section_code: section[:code], student_user_id: current_user.id), confirm: t('section.confirm.remove'), method: :post)}.join(', ')
+            - else
+              = raw current_user_sections_as_student.map {|section| t('section.code_with_teacher', section_code: section.code, teacher_name: section.teacher.name)}.join(', ')
```
Comment (Hamms) [inline dashboard/app/views/home/index.html.haml:73]: can either of these lines be turned into multiple lines?

File: dashboard/app/controllers/followers_controller.rb (hunk 1 above)
Comment (ashercodeorg) [reply dashboard/app/controllers/followers_controller.rb]: Right. Thanks.

File: dashboard/app/views/home/index.html.haml (hunk 2 above)
Comment (ashercodeorg) [reply dashboard/app/views/home/index.html.haml:73]: Not cleanly, as I don't think separating after a comma increases readability.

**Comment by ashercodeorg** [comment]
FYI @davidwufer.

**Pull request merged**